src/tests/Model/TestModelTests.cs

```diff
@@ -37,12 +38,14 @@ public class TestModelTests
         [OneTimeSetUp]
         public void CreateTestModel()
         {
-            var engine = TestEngineActivator.CreateInstance();
-            Assert.NotNull(engine, "Unable to create engine instance for testing");
+            using (var engine = TestEngineActivator.CreateInstance())
```

Comment:
This looks like a potential bug. We create an engine and give it to the model. The model continues to exist after the method exits and is used by the rest of the tests. It holds and uses the engine, which has already been Disposed on exit from the using. 

Apparently, we don't use anything in the current tests that gets disposed, or they wouldn't be passing. But we shouldn't Dispose the engine directly at all, since the model controls it's lifetime.